src/main/java/com/google/sps/servlets/UserFoldersServlet.java

```diff
@@ -35,29 +37,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     
     UserService userService = UserServiceFactory.getUserService();
 
-    if (userService.isUserLoggedIn()) {
+    Map<String, Object> jsonInfo = new HashMap<>();
+    jsonInfo.put("showCreateFormStatus", false);
+    List<Folder> userFolders = new ArrayList<>();
 
-      List<Folder> userFolders = new ArrayList<>();
+    if (userService.isUserLoggedIn()) {
 
+      jsonInfo.put("showCreateFormStatus", true);
       String userKey = request.getParameter("userKey");
 
-      // Query all folders identified by the userKey
       Query folderQuery = new Query("Folder").setAncestor(KeyFactory.stringToKey(userKey));
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       PreparedQuery results = datastore.prepare(folderQuery);
 
-      // Add folders to a list
       if (results != null) {
         for (Entity entity: results.asIterable()) {
-          Folder folder = Folder.EntityToFolder(entity);
+          Folder folder = new Folder(entity);
+          folder.setFolderKey(KeyFactory.keyToString(entity.getKey()));
           userFolders.add(folder);
         }
       }
-      
-      response.setContentType("application/json;");
-      response.getWriter().println(new Gson().toJson(userFolders));
     }
+
+    jsonInfo.put("userFolders", userFolders);
```

Comment:
I would normally suggest using a constant here as well, but there's only one instance so not too big of a deal.

In general, if a string literal is used in multiple places and should be kept in sync, you should use a constant. It makes the literal easier to change and avoid bugs because the compiler will catch the typo for you.